# Comparison of slideshow creating methods

In [2]:
from usefull_functions import (total_score,
     create_vertical_slides,
     load_photos_from_json)
from vertical_photos_combining_methods import  (random_pair_vertical_photos,
     similar_pair_vertical_photos,
     different_pair_vertical_photos)
from slideshow_creating_methods import (random_slideshow,
                                        nearest_neighbor_slideshow,
                                        grouped_slideshow,
                                        mixed_slideshow)
import random
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

## Random method

### Random Vertical photos

In [3]:
v_photos = load_photos_from_json("../data/vertical_photos.json")
h_slides = load_photos_from_json("../data/horizontal_photos.json")
v_slides = create_vertical_slides(random_pair_vertical_photos, v_photos)
slides = v_slides + h_slides
slideshow = random_slideshow(slides)
score = total_score(slideshow)
print(score)

174829


### Similar Vertical photos

In [4]:
v_photos = load_photos_from_json("../data/vertical_photos.json")
h_slides = load_photos_from_json("../data/horizontal_photos.json")
v_slides = create_vertical_slides(similar_pair_vertical_photos, v_photos)
slides = v_slides + h_slides
slideshow = random_slideshow(slides)
score = total_score(slideshow)
print(score)

141494


### Different Vertical photos

In [5]:
v_photos = load_photos_from_json("../data/vertical_photos.json")
h_slides = load_photos_from_json("../data/horizontal_photos.json")
v_slides = create_vertical_slides(different_pair_vertical_photos, v_photos)
slides = v_slides + h_slides
slideshow = random_slideshow(slides)
score = total_score(slideshow)
print(score)

187609


## Nearest neighbor (greedy) method

### Random Vertical photos

In [6]:
v_photos = load_photos_from_json("../data/vertical_photos.json")
h_slides = load_photos_from_json("../data/horizontal_photos.json")
v_slides = create_vertical_slides(random_pair_vertical_photos, v_photos)
slides = v_slides + h_slides
slideshow = nearest_neighbor_slideshow(slides)
score = total_score(slideshow)
print(score)

342730


### Similar Vertical photos

In [7]:
v_photos = load_photos_from_json("../data/vertical_photos.json")
h_slides = load_photos_from_json("../data/horizontal_photos.json")
v_slides = create_vertical_slides(similar_pair_vertical_photos, v_photos)
slides = v_slides + h_slides
slideshow = nearest_neighbor_slideshow(slides)
score = total_score(slideshow)
print(score)

302256


### Different Vertical photos

In [8]:
v_photos = load_photos_from_json("../data/vertical_photos.json")
h_slides = load_photos_from_json("../data/horizontal_photos.json")
v_slides = create_vertical_slides(different_pair_vertical_photos, v_photos)
slides = v_slides + h_slides
slideshow = nearest_neighbor_slideshow(slides)
score = total_score(slideshow)
print(score)

371787


## Grouping by tags

### Random Vertical photos

In [9]:
v_photos = load_photos_from_json("../data/vertical_photos.json")
h_slides = load_photos_from_json("../data/horizontal_photos.json")
v_slides = create_vertical_slides(random_pair_vertical_photos, v_photos)
slides = v_slides + h_slides
slideshow = grouped_slideshow(slides)
score = total_score(slideshow)
print(score)

224602


### Similar Vertical photos

In [10]:
v_photos = load_photos_from_json("../data/vertical_photos.json")
h_slides = load_photos_from_json("../data/horizontal_photos.json")
v_slides = create_vertical_slides(similar_pair_vertical_photos, v_photos)
slides = v_slides + h_slides
slideshow = grouped_slideshow(slides)
score = total_score(slideshow)
print(score)

192229


### Different Vertical photos

In [11]:
v_photos = load_photos_from_json("../data/vertical_photos.json")
h_slides = load_photos_from_json("../data/horizontal_photos.json")
v_slides = create_vertical_slides(different_pair_vertical_photos, v_photos)
slides = v_slides + h_slides
slideshow = grouped_slideshow(slides)
score = total_score(slideshow)
print(score)

241378


## Mixed method

### Random Vertical photos

In [12]:
v_photos = load_photos_from_json("../data/vertical_photos.json")
h_slides = load_photos_from_json("../data/horizontal_photos.json")
v_slides = create_vertical_slides(random_pair_vertical_photos, v_photos)
slides = v_slides + h_slides
slideshow = mixed_slideshow(slides)
score = total_score(slideshow)
print(score)

385073


### Similar Vertical photos

In [13]:
v_photos = load_photos_from_json("../data/vertical_photos.json")
h_slides = load_photos_from_json("../data/horizontal_photos.json")
v_slides = create_vertical_slides(similar_pair_vertical_photos, v_photos)
slides = v_slides + h_slides
slideshow = mixed_slideshow(slides)
score = total_score(slideshow)
print(score)

346490


### Different Vertical photos

In [14]:
v_photos = load_photos_from_json("../data/vertical_photos.json")
h_slides = load_photos_from_json("../data/horizontal_photos.json")
v_slides = create_vertical_slides(different_pair_vertical_photos, v_photos)
slides = v_slides + h_slides
slideshow = mixed_slideshow(slides)
score = total_score(slideshow)
print(score)

414179


## Compare afficiency of all methods

|           | Random | Nearest neighbor | Grouped by tags | Mixed  |
|-----------|--------|------------------|-----------------|--------|
| Random    | 174829 | 342730           | 224602          | 385073 |
| Similar   | 141494 | 302256           | 192229          | 346490 |
| Different | 187609 | 371787           | 241378          | 414179 |

Max result - 415180